### PR TITLE
Fix eth verifier and lc

### PR DESCRIPTION
**Pull request by aeryz**

### CODE REVIEW

File: light-clients/ethereum-light-client/src/client.rs [hunk 1]
```diff
@@ -104,195 +102,45 @@ impl IbcClient for EthereumLightClient {
...
-        )?;
 
-        let proof_data = header.account_update.account_proof;
-
-        verify_account_storage_root(
-            header.consensus_update.attested_header.execution.state_root,
-            &wasm_client_state.data.ibc_contract_address,
-            &proof_data.proof,
-            &proof_data.storage_root,
+        verify_header::<beacon_api_types::Mainnet>(
```
[the remark this answers is not in the document]
Comment (aeryz) [reply light-clients/ethereum-light-client/src/client.rs:112]: branching stuff is left to the other pr where i will change the protos and remove unnecessary fields etc

File: light-clients/ethereum-light-client/src/client.rs [hunk 2]
```diff
@@ -104,195 +102,45 @@ impl IbcClient for EthereumLightClient {
...
-        )
-        .map_err(Error::ValidateLightClient)?;
-
-        validate_light_client_update::<LightClientContext<Config>, VerificationContext>(
-            &ctx,
-            misbehaviour.update_2,
-            current_slot,
-            wasm_client_state.data.genesis_validators_root,
-            VerificationContext { deps },
+        verify_misbehaviour::<beacon_api_types::Mainnet>(
```
[the remark this answers is not in the document]
Comment (aeryz) [reply light-clients/ethereum-light-client/src/client.rs:128]: same response here

File: light-clients/ethereum-light-client/types/src/light_client_update_data.rs [hunk 3]
```diff
@@ -15,6 +18,24 @@ pub struct LightClientUpdateData {
     pub signature_slot: u64,
 }
 
+impl LightClientUpdateData {
+    pub fn new_beacon_light_client_update(
```
[the remark this answers is not in the document]
Comment (aeryz) [reply light-clients/ethereum-light-client/types/src/light_client_update_data.rs:22]: correct

[the remark this answers is not in the document]
Comment (aeryz) [reply light-clients/ethereum-light-client/types/src/light_client_update_data.rs:22]: will fix this in the new pr after this

**Review by benluelo — APPROVED** [review]

**Pull request merged**